coriolis/tests/osmorphing/test_debian.py

```diff
@@ -0,0 +1,297 @@
+                    }
+                }
+            }
+        }
+
+        result = self.morpher._compose_netplan_cfg(self.nics_info)
+
+        mock_yaml_dump.assert_called_once_with(
+            expected_cfg, default_flow_style=False)
+
+        self.assertEqual(result, mock_yaml_dump.return_value)
+
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_write_file_sudo')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_exec_cmd_chroot')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_test_path')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_compose_netplan_cfg')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_list_dir')
+    @mock.patch.object(
+        debian.BaseDebianMorphingTools, 'disable_predictable_nic_names'
+    )
+    @mock.patch.object(
+        debian.BaseDebianMorphingTools, '_compose_interfaces_config'
+    )
+    def test_set_net_config(
+            self, mock_compose_interfaces_config,
+            mock_disable_predictable_nic_names, mock_list_dir,
+            mock_compose_netplan_cfg, mock_test_path, mock_exec_cmd_chroot,
+            mock_write_file_sudo):
+
+        dhcp = True
+        mock_test_path.return_value = True
+        mock_list_dir.return_value = ['file1.yaml', 'file2.yml']
+
+        self.morpher.set_net_config(self.nics_info, dhcp)
+
+        mock_disable_predictable_nic_names.assert_called_once()
+        mock_test_path.assert_has_calls([
+            mock.call('etc/network'),
+            mock.call('etc/network/interfaces'),
+            mock.call('etc/netplan')])
+        mock_list_dir.assert_called_once_with('etc/netplan')
+        mock_compose_netplan_cfg.assert_called_once_with(self.nics_info)
+        mock_compose_interfaces_config.assert_called_once_with(self.nics_info)
+        mock_write_file_sudo.assert_has_calls(
+            [mock.call('etc/network/interfaces',
+                       mock_compose_interfaces_config.return_value),
+             mock.call('etc/netplan/coriolis_netplan.yaml',
+                       mock_compose_netplan_cfg.return_value)])
+        mock_exec_cmd_chroot.assert_has_calls(
+            [mock.call('cp etc/network/interfaces etc/network/interfaces.bak'),
+             mock.call('mv etc/netplan/file1.yaml etc/netplan/file1.yaml.bak'),
+             mock.call('mv etc/netplan/file2.yml etc/netplan/file2.yml.bak')])
+
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_write_file_sudo')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_exec_cmd_chroot')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_test_path')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_compose_netplan_cfg')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_list_dir')
+    @mock.patch.object(
+        debian.BaseDebianMorphingTools, 'disable_predictable_nic_names'
+    )
+    @mock.patch.object(
+        debian.BaseDebianMorphingTools, '_compose_interfaces_config'
+    )
+    def test_set_net_config_no_dhcp(
```

Comment:
Same suggestion as above, avoid mocking `_compose_netplan_cfg` and `_compose_interfaces_config`.